pulsar-client/src/main/java/com/yahoo/pulsar/client/api/Consumer.java

```diff
@@ -231,4 +233,12 @@
      * breaks, the messages are redelivered after reconnect.
      */
     void redeliverUnacknowledgedMessages();
+
+    /**
+     * Redelivers the given unacknowledged messages. In Failover mode, the request is ignored if the consumer is not
+     * active for the given topic. In Shared mode, the consumers messages to be redelivered are distributed across all
+     * the connected consumers. This is a non blocking call and doesn't throw an exception. In case the connection
+     * breaks, the messages are redelivered after reconnect.
+     */
+    void redeliverUnacknowledgedMessages(List<MessageIdImpl> messageIds);
```

Comment:
> What would be the use case for exposing this in the public API? I'm not against it, but once it's out, we cannot remove it, so we should only add it if it's really needed. 
Also, the method should be called redeliverMessages(messageIds)ç

I believe there would be no need for exposing this, at least not now, but maybe we could make `void redeliverUnacknowledgedMessages()` have the default behavior of not redelivering all messages.
I think this should not be called `redeliverMessages(messageIds)` because the broker actually makes sure it is redelivering only unacknowledged messages.

> And we should take MessageId and not MessageIdImpl

I've seen a lot of things using MessageIdImpl, that's why I went that way, it's a trivial change.